lib/multi_aggregator/processor.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module MultiAggregator
+  class Processor
+    Error = Class.new(::MultiAggregator::Error)
+
+    attr_reader(
+      :storage_validator,
+      :data_transfer,
+      :logger,
+      :query_validator,
+      :spec_creator
+    )
+
+    def initialize(
+      storage_validator: MultiAggregator::Adapters::Validator.new,
```

Comment:
Could you elaborate more please? Are trying to implement DI this here?
@Olefine 's point that we don't need this in `initialize`